terraform/modules/bap_env_gcp/variables.tf

```diff
@@ -49,9 +54,15 @@ variable "mariadb_machine_image" {
   default = "debian-cloud/debian-11"
 }
 
+variable "mariadb_disk_auto_delete" {
```

Comment:
We only need one variable and not one per instance. It's either you want persistent disks or not. I don't see the point of configuring the ones that need it with a specific variable.